bench/bench_htbl.ml

```diff
@@ -0,0 +1,83 @@
+open Multicore_bench
+
+module Key = struct
+  type t = int
+
+  let equal = Int.equal
+  let hash = Fun.id
+end
+
+let run_one ~budgetf ~n_domains ?(n_ops = 20 * Util.iter_factor)
+    ?(n_keys = 10000) ~percent_mem ?(percent_add = (100 - percent_mem + 1) / 2)
+    ?(prepopulate = true) (module Htbl : Htbl_intf.HTBL) =
+  let percent_rem = 100 - (percent_mem + percent_add) in
+
+  let limit_mem = percent_mem in
+  let limit_add = percent_mem + percent_add in
+
+  assert (0 <= limit_mem && limit_mem <= 100);
+  assert (limit_mem <= limit_add && limit_add <= 100);
+
+  let t = Htbl.create ~hashed_type:(module Key) () in
+  if prepopulate then
+    for _ = 1 to n_keys do
+      let value = Random.bits () in
+      let key = value mod n_keys in
+      Htbl.try_add t key value |> ignore
+    done;
+
+  let n_ops = (100 + percent_mem) * n_ops / 100 in
+  let n_ops = n_ops * n_domains in
+
+  let n_ops_todo = Atomic.make 0 |> Multicore_magic.copy_as_padded in
+
+  let init _ =
+    Atomic.set n_ops_todo n_ops;
+    Random.State.make_self_init ()
+  in
+  let work _ state =
+    let rec work () =
+      let n = Util.alloc n_ops_todo in
+      if n <> 0 then
+        let rec loop n =
+          if 0 < n then
+            let value = Random.State.bits state in
+            let op = (value asr 20) mod 100 in
+            let key = value mod n_keys in
+            if op < limit_mem then begin
+              Htbl.mem t key |> ignore;
+              loop (n - 1)
+            end
+            else if op < limit_add then begin
+              Htbl.try_add t key value |> ignore;
+              loop (n - 1)
+            end
+            else begin
+              Htbl.try_remove t key |> ignore;
+              loop (n - 1)
+            end
+          else work ()
+        in
+        loop n
+    in
+    work ()
+  in
+
+  let config =
+    Printf.sprintf "%d workers, %d%% mem %d%% add %d%% rem" n_domains
+      percent_mem percent_add percent_rem
+  in
```

Comment:
![Screenshot 2024-10-10 at 15 21 59](https://github.com/user-attachments/assets/ce489d1a-b55d-4e56-a98c-d04a7dcf292b)

This coloring is due to not having unique configs.

Reply:
I'll look into adding a validation step into `multicore-bench` that would raise an error when this happens.  It shouldn't be too difficult.

Reply:
I have [a PR in multicore-bench](https://github.com/ocaml-multicore/multicore-bench/pull/29) where duplicates like this will raise a failure that I'll merge soon and publish a new version of multicore-bench.

Reply:
`multicore-bench` now checks that there are no duplicates and fails the bench if there are.